### DISCUSSION TITLE
Array of arrays state best practices pattern?

### DISCUSSION THREAD

**Discussion by pabloruizetc** [vanjs-org/van#183] [Q&A] [answered]
I'm currently working on a web-based video editor and I'm seeking some advice on how to best model the timeline state. In this editor, users can work with different types of tracks such as video, audio, and text tracks. Each track type has a set of clips that define specific segments within each track.

I tried to use `vanX.reactive` for the tracks state and then use vanX.list, but when I have to render the array of clips inside each track I don't know what's the best way to proceed. Maybe use `.map` directly?

**Comment by Tao-VanJS** [comment, marked as answer]
I think you can use another inner [`vanX.list`](https://vanjs.org/x#reactive-list) for the array of clips.
